scripts/calculated_interventions.js

```diff
@@ -1,102 +1,55 @@
 const _ = require('lodash');
-const axios = require('axios');
 const fs = require('fs-extra');
-const Promise = require('bluebird');
 const path = require('path');
-const { Projections } = require('./../models');
-const { US_STATES } = require('./../enums');
-const DataUrlJson = require('../../src/assets/data/data_url.json');
-
-const DATA_URL = DataUrlJson.data_url.replace(/\/$/, '');
-
-async function getStateAndCountyDataFiles(stateCode) {
-  const files = [
-    `${stateCode}.0.json`,
-    `${stateCode}.1.json`,
-    `${stateCode}.3.json`,
-    // `${stateCode}.2.json`,
-  ];
-
-  const summaryFile = `county_summaries/${stateCode}.summary.json`;
-
-  try {
-    const countyFipsData = {};
-    const stateData = await Promise.mapSeries(files, async file =>
-      fetchJson(file),
-    );
-    const stateProjections = new Projections(stateData, stateCode);
-
-    const stateSummaryData = await fetchJson(summaryFile);
-
-    await Promise.map(
-      stateSummaryData.counties_with_data,
-      async fipsCode => {
-        const files = [
-          `county/${stateCode}.${fipsCode}.0.json`,
-          `county/${stateCode}.${fipsCode}.1.json`,
-          `county/${stateCode}.${fipsCode}.3.json`,
-          // `county/${stateCode}.${fipsCode}.2.json`,
-        ];
-
-        try {
-          const countyData = await Promise.mapSeries(files, async file =>
-            fetchJson(file),
-          );
-          const countyProjections = new Projections(countyData, stateCode);
-
-          countyFipsData[
-            fipsCode
-          ] = countyProjections.getThresholdInterventionLevel();
-        } catch (err) {
-          console.error(`Failed to get data for ${stateCode}.${fipsCode}`, err);
-        }
-      },
-      { concurrency: 100 },
-    );
-
-    return {
-      stateProjections,
-      stateInterventionColor: stateProjections.getThresholdInterventionLevel(),
-      countyFipsData,
-    };
-  } catch (err) {
-    console.error(`Failed to get data for ${stateCode}`, err);
-  }
-}
+const US_STATES = require('./../src/enums/us_states');
+import { fetchStateSummary, fetchProjections } from '../src/utils/model.js';
+
+//const DataUrlJson = require('./../src/assets/data/data_url.json');
+//const DATA_URL = DataUrlJson.data_url.replace(/\/$/, '');
+
+async function getStateAndCountyDataFiles(stateAbbr) {
+  const stateSummaryData = await fetchStateSummary(stateAbbr);
+  const stateProjections = await fetchProjections(stateAbbr);
+
+  let countyFipsData = {};
+  await Promise.all(
+    stateSummaryData.counties_with_data.map(async fipsCode => {
+      try {
+        const countyProjections = await fetchProjections(stateAbbr, {
+          full_fips_code: fipsCode,
+        });
+        countyFipsData[fipsCode] = countyProjections;
+      } catch (ex) {
+        console.error(ex);
+      }
+    }),
```

Comment:
huhh i guess would this be slow?

Reply:
slower than reading from the filesystem, but all parallelized